src/App.js

```diff
@@ -1,48 +1,90 @@
 import React from "react";
 import TodoList from "./components/TodoComponents/TodoList";
 import TodoForm from "./components/TodoComponents/TodoForm";
+import "./css/main.css";
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
   constructor() {
     super();
-    this.state = {
-      taskList: [],
-      task: ""
-    };
+    // Retrieved stringified state object from localStorage (if any):
+    const retrievedState = localStorage.getItem("storedState");
+    // Parses state object back to its original formatting:
+    const parsedState = JSON.parse(retrievedState);
+    // Uses state object from localStorage if there is one. If not uses a default state object:
+    parsedState
+      ? (this.state = parsedState)
+      : (this.state = {
+          taskList: [],
+          task: ""
+        });
   }
 
+  // Handles changes in the form input box:
   handleChanges = e => {
-    console.log(`handleChanges: ${e.target.value}`);
     this.setState({
       [e.target.name]: e.target.value
     });
   };
 
+  // Adds a task to the list when a user clicks the "Add Task" button or presses enter:
   addTask = e => {
     e.preventDefault();
+    // User cannot add an empty string to the to-do list:
+    if (!this.state.task) {
+      alert(
+        `If you want to do nothing, just do nothing.\nDon't add "nothing" to your to-do list.`
```

Comment:
love the custom alert message: this can also be handled with less code (but without a custom message) by using the `required` attribute on the input field